docs/source/howto/forced-login.md

```diff
@@ -0,0 +1,129 @@
+This is [disabled by default](#HubAuth.allow_token_in_url) in JupyterHub 5, because it presents a vulnerability for users to craft links that let _other_ users login as them, which can lead to inter-user attacks.
+
+But that leaves the question: how do I as an _application developer_ embedding JupyterHub link users to their own running server without triggering another login prompt?
+
+The problem with `?token=...` in the URL is specifically that _users_ can get and create these tokens, and share URLs.
+This wouldn't be an issue if only authorized applications could issue tokens that behave this way.
+The single-user server doesn't exactly have the hooks to manage this easily, but the [Authenticator](#Authenticator) API does.
+
+## Problem statement
+
+We want our external application to be able to:
+
+1. authenticate users
+2. (maybe) create JupyterHub users
+3. start JupyterHub servers
+4. redirect users into running servers _without_ any login prompts/loading pages from JupyterHub, and without any prior JupyterHub credentials
+
+Step 1 is up to the application and not JupyterHub's problem.
+Step 2 and 3 use the JupyterHub [REST API](#jupyterhub-rest-API).
+The service would need the scopes:
+
+```
+admin:users # creating users
+servers # start/stop servers
+```
+
+That leaves the last step: sending users to their running server with credentials, without prompting login.
+This is where things can get tricky!
+
+### Ideal case: oauth
+
+_Ideally_, the best way to set this up is with the external service as an OAuth provider,
+though in some cases it works best to use proxy-based authentication like Shibboleth / [REMOTE_USER](https://github.com/cwaldbieser/jhub_remote_user_authenticator).
+The main things to know are:
+
+- Links to `/hub/user-redirect/some/path` will ultimately land users at `/users/theirserver/some/path` after completing login, ensuring the server is running, etc.
+- Setting `Authenticator.auto_login = True` allows beginning the login process without JupyterHub's "Login with..." prompt
+
+_If_ your OAuth provider allows logging in to external services via your oauth provider without prompting, this is enough.
+Not all do, though.
+
+If you've already ensured the server is running, this will _appear_ to the user as if they are being sent directly to their running server.
+But what _actually_ happens is quite a series of redirects, state checks, and cookie-setting:
+
+1. visiting `/hub/user-redirect/some/path` checks if the user is logged in
+   1. if not, begin the login process (`/hub/login?next=/hub/user-redirect/...`)
+   2. redirects to your oauth provider to authenticate the user
+   3. redirects back to `/hub/oauth_callback` to complete login
+   4. redirects back to `/hub/user-redirect/...`
+2. once authenticated, checks that the user's server is running
+   a. if not running, begins launch of the server and redirects to `/hub/spawn-pending/?next=...`
```

Comment:
Nested numeric bullets look fine, nested letters are merged into the previous line:
![image](https://github.com/user-attachments/assets/cddd4030-0ac4-4bef-825e-5022106daabf)